### ISSUE TITLE
[BUG] MaskedDataset.__getitem__ clones x instead of y for y_masked

### ISSUE THREAD

**Issue by Alleny244** [gc-os-ai/pyaptamer#485]
**Describe the bug**
                                                                                                                                                                                                                                                                                          
  In MaskedDataset.__getitem__, line 154 clones x instead of y when creating y_masked. This means the training target for masked language modeling contains values from the input sequence rather than the target sequence.                                                                
   


**To Reproduce**

```python
                                                                                                                                                                                                                                                                                        
  from pyaptamer.datasets.dataclasses import MaskedDataset
                                                                                                                                                                                                                                                                                           
  x = [[1, 2, 3, 4, 0]]
  y = [[5, 6, 7, 8, 0]]  # deliberately different from x                                                                                                                                                                                                                                   
  ds = MaskedDataset(x, y, max_len=5, mask_idx=9, masked_rate=0.2)                                                                                                                                                                                                                         
   
  x_masked, y_masked, x_orig, y_orig = ds[0]                                                                                                                                                                                                                                               
                                                                                                                                                                                                                                                                                         
  # y_masked contains values from x (1,2,3,4), not from y (5,6,7,8)                                                                                                                                                                                                                        
  print(y_masked)  # shows values from x instead of y                                                                                                                                                                                                                                    
                                                                   
```

**Expected behavior**
 y_masked should be derived from y, not x. Line 154 should be:                                                                                                                                                                                                                            
  y_masked = y.clone().detach()
  instead of:                                                                                                                                                                                                                                                                              
  y_masked = x.clone().detach()                                                                                                                                                                                                                                                            
                                         

**Additional context**
  File: pyaptamer/datasets/dataclasses/_masked.py, line 154.  
 y_masked should be derived from y, not x. Line 154 should be:                                                                                                                                                                                                                            
  y_masked = y.clone().detach()
  instead of:                                                                                                                                                                                                                                                                              
  y_masked = x.clone().detach()                                                                                                                                                                                                                                                            
                                         

**Versions**
<details>

  pyaptamer 0.1.0a1  

</details>
